File: Learn/maxelement/maxelem.go
package main
import (
    "fmt"
)

func max_element_position(a []int, size int) (int, int) {
    positionOfMaxElement := 0
    if size < 1 { return -1, -1 }
    for i := 0; i < size; i++ {
        if a[positionOfMaxElement] < a[i] {
            positionOfMaxElement = i
        }
    }
    return positionOfMaxElement, a[positionOfMaxElement]
}

func main() {
    var size, pos, value, i int
    a := make([]int, 10)
    fmt.Scanf("%d", &size)
    for i = 0; i < size; i++ {
        fmt.Scanf("%d", &a[i])
    }
    pos, value = max_element_position(a, size)
    if pos != -1 {
        fmt.Printf("%d\n%d", pos, value);
    }
}

File: Learn/maxelement/maxelem.cpp
/**
 * CPP alternative to find max element from an array
 */
#include <iostream>
using namespace std;

int max_element_position(int a[], int size) {
    int i, positionOfMaxElement = 0;
    if(size < 1) return -1;
    for(i = 0; i < size; ++i)
        if(a[i] > a[positionOfMaxElement])
            positionOfMaxElement = i;
    return positionOfMaxElement;
}

int main() {
    int a[10], size = 0, i, pos;
    cin >> size;
    for(i = 0; i < size; ++i) {
        cin >> a[i];
    }
    pos = max_element_position(a, size);
    if(pos == -1) return 1;
    cout << pos << "\n" << a[pos];
    return 0;
}